daffodil-lib/src/test/scala/LineCounter.scala

```diff
@@ -71,35 +86,24 @@ object LineCounter extends App {
-            "link16ed6-daffodil-compliant-original.xsd", "link16ed6-daffodil-compliant-slow.xsd", "link16ed6-daffodil-compliant.xsd", // link16 nato generated schema
-            "proposed_ded.dfdl.xsd", "proposed_msg.dfdl.xsd", "dfidui.dfdl.xsd") //
-            ),
-        new LineCounter("vmf", root + "fouo-schemas/vmf", List("generator", "schema"), List("src/main", "src/test")),
-        new LineCounterOneDir("vmf other", root + "fouo-schemas/vmf", List("build.sbt", "README.md") //
-        ))), //
+            ,
+        new LineCounter("daffodil tutorial tdml tests", root + "daffodil", List("tutorials")),
+        new LineCounter("daffodil std-layout tdml tests", root + "daffodil", List("test-stdLayout")),
+        new LineCounter("openDFDL", root + "openDFDL/examples"))),
+      new LineCounterCombiner("Public schemas", List(new LineCounter("DFDLSchemas", root + "DFDLSchemas"),
+        new LineCounter("jpeg2000", root, List("jpeg2000")),
+        new LineCounter("asterisk", root, List("asterisk")),
+        new LineCounter("ipfix", root + "fouo-schemas", List("ipfix"), List("")))),
+      new LineCounter("fouo-schemas", root + "fouo-schemas",
+        filesToExclude = List("army_drrs_lh.dfdl.tdml", // a giant tdml file that mostly just contains data
+          "link16ed6-daffodil-compliant-original.xsd", "link16ed6-daffodil-compliant-slow.xsd", "dfidui.dfdl.xsd", // link16 nato generated schema
+          "proposed_ded.dfdl.xsd", "proposed_msg.dfdl.xsd") //
+      ),
       new LineCounterCombiner("Integrations",
-        List(
-          new LineCounter("calabash proper", root + "daffodil-calabash-extension", List("calabash-server"), List("src/main", "src/test"),
-            fileSuffixesToInclude = List(".scala", ".java", ".tdml", ".xsd", ".xpl", ".sbt")),
-          new LineCounter("calabash fouo", root + "daffodil-fouo", List("calabash-test"), List("src/main", "src/test"),
-            fileSuffixesToInclude = List(".scala", ".java", ".tdml", ".xpl", ".sbt")),
-          new LineCounter("NiFi", root + "daffodil-nifi", List("nifi-daffodil-nar", "nifi-daffodil-processors"), List("src/main", "src/test")),
-          new LineCounter("Spark", root, List("daffodil-spark"), List("src/main", "src/test")))),
-      new LineCounter("all sbt files", root, List("daffodil", "fouo-schemas", "DFDLSchemas", "daffodil-calabash-extension",
-        "daffodil-nifi", "jpeg2000", "daffodil-nifi", "daffodil-spark", "ipfix", "asterix"), List(""), fileSuffixesToInclude = List(".sbt")))
+        List(new LineCounter("calabash proper", root + "daffodil-calabash-extension", List("calabash-server")),
+          new LineCounter("calabash fouo", root + "daffodil-fouo", List("calabash-test")),
+          new LineCounter("NiFi", root + "daffodil-nifi", List("nifi-daffodil-nar", "nifi-daffodil-processors")),
+          new LineCounter("Spark", root, List("daffodil-spark")))))
 
```

Comment:
This is all specific to your directory structure, and really aren't Daffodil related, aside from the fact they they're DFDL schemas. To me. this feels like this line counter should live outside this Daffodil repo and should have a config file to specicy the directories and file extensions to include/exclude. Maybe it should live in OpenDFDL?

In fact, I'm not sure if it's evn worth maintaining this. Seems effort would be better spent using the [sbt-status](https://github.com/orrsella/sbt-stats) plugin.  The looks pretty extendable to add new metrics like TDML files, DFDL schema files, etc. Then it's just a matter a running a script that runs "sbt stats" on each project and adding up the numbers.

Reply:
Created an improved sbt-stats which is at OpenDFDL/sbt-stats.
I have a pull request open to the original sbt-stats project, but for now this is available via the fork.